packages/x-charts/src/BarChart/BarLabel/barLabelClasses.tsx

```diff
@@ -0,0 +1,36 @@
+import generateUtilityClass from '@mui/utils/generateUtilityClass';
+import generateUtilityClasses from '@mui/utils/generateUtilityClasses';
+import composeClasses from '@mui/utils/composeClasses';
+import type { BarLabelOwnerState } from './types';
+
+export interface BarLabelClasses {
+  /** Styles applied to the root element. */
+  root: string;
+  /** Styles applied to the root element if it is highlighted. */
+  highlighted: string;
+  /** Styles applied to the root element if it is faded. */
+  faded: string;
+}
+
+export type BarLabelClassKey = keyof BarLabelClasses;
+
+export function getBarLabelUtilityClass(slot: string) {
+  return generateUtilityClass('MuiBarLabel', slot);
+}
+
+export const barLabelClasses = generateUtilityClasses('MuiBarLabel', [
+  'root',
+  'highlighted',
+  'faded',
+]);
+
+export const useUtilityClasses = (ownerState: BarLabelOwnerState) => {
+  const { classes, seriesId, isFaded, isHighlighted } = ownerState;
+  const slots = {
+    root: ['root', `series-${seriesId}`],
+    highlighted: [isHighlighted && 'highlighted'],
+    faded: [isFaded && 'faded'],
```

Comment:
Not directly related to this PR, because this approach is already used, but IMHO, in such cases a more correct approach would be something like this:
```suggestion
    root: ['root', `series-${seriesId}`, isHighlighted && 'highlighted', isFaded && 'faded']
```

MuiButton for reference:
https://github.com/mui/material-ui/blob/93e25916268e33832e78a10e3b822c25f6902a56/packages/mui-material/src/Button/Button.js#L21-L35

Reply:
Odd, I tried this, it was my first try since I copy/pasted the behaviour from somewhere else, but the it wasn't working. Though I had to change a lot of things for the behaviour to work as I expected, will try your change and report back. 👍